src/gobapi/dump/to_db.py

```diff
@@ -2,129 +2,249 @@
-                file=stream,
-                size=BUFFER_PER
-            )
+                if stream.total_count >= commit:
+                    connection.commit()
+                    commit += COMMIT_PER
 
-            if stream.total_count >= commit:
-                connection.commit()
-                commit += COMMIT_PER
+                    yield f"\n{self.collection_name}: {stream.total_count:,}"
+                else:
+                    # Let client know we're still working.
+                    yield "."
 
-                yield f"\n{collection_name}: {stream.total_count:,}"
-            else:
-                # Let client know we're still working.
-                yield "."
+        yield f"\nExported {stream.total_count} rows\n"
+        connection.commit()
 
-    yield(f"\nExported {stream.total_count} rows\n")
-    connection.commit()
+    def _filter_last_events_lambda(self, max_eventid):
+        return lambda table: getattr(table, FIELD.LAST_EVENT) > max_eventid
 
-    yield f"Finalize table {collection_name}\n"
-    rename_table = _rename_table(schema, current_name=tmp_collection_name, new_name=collection_name)
-    result = engine.execute(rename_table)
-    result.close()
+    def dump_to_db(self, full_dump=False):
+        """Runs dump for this instance. Tries to synchronise based on last event by default. Set full_dump=True to
+        replace all existing data and ignore synchronisation based on events.
+        """
+        yield from self._prepare_destination()
 
-    yield "Create default indexes\n"
-    yield from _create_indexes(engine, schema, collection_name, model)
+        max_eventid = None if full_dump else self._max_eventid_dst()
 
-    connection.close()
+        if max_eventid is not None and self._table_columns_equal(self.collection_name, self.tmp_collection_name):
+            yield "Have earlier dump and columns have not changed. Do sync dump\n"
+            # Already have records in destination table, and model has not changed. Do sync dump
 
+            self._copy_table_into(self.collection_name, self.tmp_collection_name)
 
-def dump_to_db(catalog_name, collection_name, config):
-    engine = create_engine(URL(**config['db']))
-    config['engine'] = engine
+            source_ids_to_update = get_entity_refs_after(self.catalog_name, self.collection_name, max_eventid)
 
-    schema = config.get("schema", catalog_name)
-    yield f"Export {catalog_name} {collection_name} in schema {schema}\n"
+            if source_ids_to_update:
+                self._delete_dst_entities(self.tmp_collection_name, source_ids_to_update)
 
-    try:
-        entities, model = dump_entities(catalog_name, collection_name)
-        yield from _dump_to_db(schema, catalog_name, collection_name, entities, model, config)
+            entities, model = dump_entities(self.catalog_name, self.collection_name,
+                                            self._filter_last_events_lambda(max_eventid))
+        else:
```

Comment:
isn't table: getattr(table, FIELD.LAST_EVENT) > max_eventid more readable than a function that returns this lambda?

Reply:
Yes it is. Extracted the lambda into a method because of testability.